core/lib/dal/src/events_web3_dal.rs

```diff
@@ -119,10 +121,10 @@ impl EventsWeb3Dal<'_, '_> {
 
         let mut where_sql = format!("(miniblock_number >= {})", filter.from_block.0 as i64);
 
-        if let Some(to_block) = filter.to_block {
-            let block_sql = web3_block_number_to_sql(to_block);
-            where_sql += &format!(" AND (miniblock_number <= {})", block_sql);
-        }
+        let block_sql =
+            web3_block_number_to_sql(BlockNumber::Number(U64::from(filter.to_block.as_u64())));
+        where_sql += &format!(" AND (miniblock_number <= {})", block_sql);
```

Comment:
Can't you treat `filter.to_block` in the same way as `from_block` above? Conversion to `BlockNumber` looks redundant. Also, maybe off-topic, but it seems reasonable to make from / to numbers query arguments, instead of inlining them as numbers.